### PR TITLE
Task/graphql support

### PR DESCRIPTION
**Pull request by yaroslavsudnik**

### CODE REVIEW

**Review by ruslan-fayzullin — APPROVED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -30,7 +30,7 @@ android {
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         testInstrumentationRunnerArgument "runnerBuilder",
                 "de.mannodermaus.junit5.AndroidJUnit5Builder"
-        buildConfigField 'String', 'BASE_URL', "\"https://api.com\""
+        buildConfigField 'String', 'GQL_URL', "\"https://rails-base-graphql-api.herokuapp.com/graphql\""
```
Comment (IlyaEremin) [inline app/build.gradle:33]: You can leave `BASE_URL`

File: app/build.gradle [hunk 2]
```diff
@@ -81,21 +81,33 @@ android {
...
+}
+
 repositories {
     maven { url = uri("https://dl.bintray.com/kodein-framework/Kodein-DI") }
 }
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$versions.kotlin"
+    implementation 'com.google.code.gson:gson:2.8.6'
```
Comment (IlyaEremin) [inline app/build.gradle:107]: Probably can be removed

File: app/src/main/java/com/flatstack/android/di/modules/netModule.kt [hunk 3]
```diff
@@ -1,93 +1,28 @@
...
 import org.kodein.di.generic.bind
 import org.kodein.di.generic.instance
 import org.kodein.di.generic.singleton
 import org.kodein.di.generic.with
-import retrofit2.Retrofit
-import retrofit2.converter.gson.GsonConverterFactory
 
 val netModule = Kodein.Module(name = "netModule") {
-    constant("baseUrl") with BuildConfig.BASE_URL
+    constant("baseUrl") with BuildConfig.GQL_URL
```
Comment (IlyaEremin) [inline app/src/main/java/com/flatstack/android/di/modules/netModule.kt:14]: Move `"baseUrl"` to const

File: app/src/main/java/com/flatstack/android/login/LoginMapper.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.flatstack.android.login
+
+import com.flatstack.android.graphql.mutation.LoginMutation
+import com.flatstack.android.model.entities.Session
+
+object LoginMapper {
```
Comment (IlyaEremin) [inline app/src/main/java/com/flatstack/android/login/LoginMapper.kt:6]: Add test

Comment (IlyaEremin) [reply app/src/main/java/com/flatstack/android/login/LoginMapper.kt:6]: Make it class, and inject by DI

File: app/src/main/java/com/flatstack/android/model/network/errors/ErrorHandler.kt [hunk 5]
```diff
@@ -1,42 +1,48 @@
...
+    private fun userMessage(error: Error) = when (mapToStatus(error)) {
+        HttpURLConnection.HTTP_NOT_MODIFIED -> stringResource.getString(R.string.not_modified_error)
+        HttpURLConnection.HTTP_BAD_REQUEST -> stringResource.getString(R.string.bad_request_error)
+        HttpURLConnection.HTTP_UNAUTHORIZED -> stringResource.getString(R.string.unauthorized_error)
+        HttpURLConnection.HTTP_FORBIDDEN -> stringResource.getString(R.string.forbidden_error)
+        HttpURLConnection.HTTP_NOT_FOUND -> stringResource.getString(R.string.not_found_error)
+        HttpURLConnection.HTTP_BAD_METHOD -> stringResource.getString(R.string.method_not_allowed_error)
+        HttpURLConnection.HTTP_CONFLICT -> stringResource.getString(R.string.conflict_error)
+        HttpURLConnection.HTTP_INTERNAL_ERROR -> stringResource.getString(R.string.server_error_error)
+        else -> error.message
```
Comment (IlyaEremin) [inline app/src/main/java/com/flatstack/android/model/network/errors/ErrorHandler.kt:43]: Should we prefer server message over parsed messages?

File: app/src/main/java/com/flatstack/android/profile/ProfileViewModel.kt [hunk 6]
```diff
@@ -18,11 +18,7 @@ class ProfileViewModel(
         loginRepository.onDestroy()
     }
 
-    fun updateProfile() {
-        profileBoundResource.fetchFromNetwork()
-    }
+    fun updateProfile() = profileBoundResource.fetchFromNetwork()
```
Comment (IlyaEremin) [inline app/src/main/java/com/flatstack/android/profile/ProfileViewModel.kt:21]: Let's read on best practices of how to use specify return value in such blocks. I mean it's not clear does this function return something, or is it just function call.

**Review by IliaEremin — CHANGES_REQUESTED** [review]

**Comment by IlyaEremin** [comment]
@yaroslavsudnik also detekt, linter and unit tests are failed.

File: app/build.gradle (hunk 2 above)
Comment (yaroslavsudnik) [reply app/build.gradle:107]: gson using in com.flatstack.android.util.storage.Storage

File: app/src/test/java/com/flatstack/android/login/LoginMapperTest.kt [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.jupiter.api.Assertions.assertEquals
+
+internal class LoginMapperTest {
+    @Test
+    fun mapLogin() {
+        val expectedAccessToken = "access_token"
+        val expectedSignin = LoginMutation.Signin(accessToken = expectedAccessToken)
+        val expectedSession = Session(accessToken = expectedAccessToken)
+
+        assertEquals(LoginMapper().mapLogin(expectedSignin), expectedSession)
```
Comment (IlyaEremin) [inline app/src/test/java/com/flatstack/android/login/LoginMapperTest.kt:15]: Please split test into 3 sections to make obvious what method is under the test.

Comment (IlyaEremin) [reply app/src/test/java/com/flatstack/android/login/LoginMapperTest.kt:15]:
```
val actualSession = loginMapper.mapLogin(...)

assertThat(actualSession).isEqualTo(expectedSession)
```

**Review by IliaEremin — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 8]
```diff
@@ -105,9 +117,7 @@ dependencies {
     androidTestImplementation androidTestsLibs
 
     //Crashlytics
-    implementation('com.crashlytics.sdk.android:crashlytics:2.10.1@aar') {
-        transitive = true
-    }
+    implementation 'com.google.firebase:firebase-crashlytics:17.2.2'
```
[the remark this answers is not in the document]
Comment (yaroslavsudnik) [reply app/build.gradle:120]: Seems like I need to access for Firebase console

**Review by IliaEremin — APPROVED** [review]

**Pull request merged**